sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/ServicePrincipalCredentialEntity.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    /// <summary>
+    /// Authenticates to an Azure service via service principal.
+    /// </summary>
+    [CodeGenModel("ServicePrincipalCredential")]
+    [CodeGenSuppress(nameof(ServicePrincipalCredentialEntity), typeof(string), typeof(ServicePrincipalParam))]
+    public partial class ServicePrincipalCredentialEntity
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServicePrincipalCredentialEntity"/> class.
+        /// </summary>
+        /// <param name="name">A custom unique name for this <see cref="ServicePrincipalCredentialEntity"/> to be displayed on the web portal.</param>
+        /// <param name="clientId">The client ID of the service principal used for authentication.</param>
+        /// <param name="clientSecret">The client Secret of the service principal used for authentication.</param>
+        /// <param name="tenantId">The tenant ID of the service principal used for authentication.</param>
+        public ServicePrincipalCredentialEntity(string name, string clientId, string clientSecret, string tenantId) : base(name)
```

Comment:
and these values are different than the ones sued to authenticate against the client, right?

Reply:
I guess the word `client` is making me confuse... it could just be me not having enough context...

Reply:
> and these values are different than the ones sued to authenticate against the client, right?

Correct. `ClientId`, `ClientSecret`, and `TenantId` are well-known concepts used in an authentication with a Service Principal, so users should already be familiar with these terms when trying to use this credential entity. These are the same we use for AAD authentication in other services, [like in Form Recognizer](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/formrecognizer/Azure.AI.FormRecognizer#create-formrecognizerclient-with-azure-active-directory-credential).